src/services/API_configurations.ts

```diff
@@ -0,0 +1,61 @@
+import {getDoc, doc} from "firebase/firestore";
+import {db, storage} from "../modules/firebase";
+import { ref, getDownloadURL, listAll } from "firebase/storage";
+import { CarConfigurations } from "../modules/configurator/types";
+import {fileNames} from "./const/fileNames";
+
+export const fetchCarConfigurations = async (id: string) => {
+    try {
+        const configRef = doc(db, "car-models", id);
+
+        const response = await getDoc(configRef);
+        const responseData = response.data();
+
+        if (!responseData) throw new Error("Data could not be found.");
+        const carData: CarConfigurations = {
+            model: responseData.name,
+            productionYear: responseData.production_year,
+            color: responseData.color,
+            wheelVariants: responseData.wheel_variant,
+            interiorVariants: responseData.interior_variant,
+            price: responseData.price
+        }
+
+        return carData;
+    } catch (err: any) {
+        throw new Error(err);
+    }
+}
+
+export const fetchCarImagesByColorAndVariant = async (modelName: "Audi RS5" | "Audi RS6" | "Audi e-tron GT", color: string, wheelVariant: number) => {
+    try {
+        let modelFile = fileNames[modelName];
```

Comment:
Why did you use `let` here when it's not changing

Reply:
Changed to `const`.